application/tests/httproute_test.yaml

```
@@ -0,0 +1,279 @@
+suite: APIGateway-HTTPRoute
+
+templates:
+  - httproute.yaml
+
+tests:
+  - it: does not render when httpRoute is not enabled
+    set:
+      httpRoute:
+        enabled: false
+    asserts:
+      - hasDocuments:
+          count: 0
+
+  - it: does not render when Gateway API is not available
+    set:
+      httpRoute:
+        enabled: true
+    asserts:
+      - failedTemplate:
+          errorMessage: "There is no APIGateway resource definition in the target cluster!"
```

Though [the documentation speaks of therre not being a native implementation,](https://kubernetes.io/docs/concepts/services-networking/gateway/#what-s-next) it's quite possible resources under the `gateway.networking.k8s.io` group might make their way to K8s core, just like Ingress resources. It might be good to add the current K8s version under capabilities here to ensure this test job doesn't break in the distant future, what do you think?


```suggestion
  - it: does not render when Gateway API is not available
    set:
      httpRoute:
        enabled: true
    capabilities:
      majorVersion: 1
      minorVersion: 34
    asserts:
      - failedTemplate:
          errorMessage: "There is no APIGateway resource definition in the target cluster!"
```

---

good catch